core/src/helpers/parseMetamaskError.ts

```diff
@@ -1,12 +1,20 @@
 import truncateText from './truncateText';
 
-const parseMetamaskError = function (errorMessage = ''): string {
-    const jsonFirstIndex = errorMessage.indexOf('{');
-    const jsonLastIndex = errorMessage.lastIndexOf('}');
+const parseMetamaskError = function (errorMessage = ''): unknown {
+    const jsonFirstIndex = errorMessage.indexOf('"message":"');
+    if (jsonFirstIndex === -1) {
+        return truncateText(errorMessage || 'unknown');
+    }
     try {
-        const jsonString = errorMessage.substring(jsonFirstIndex, jsonLastIndex + 1);
-        const metamaskError = JSON.parse(jsonString);
-        return truncateText(metamaskError?.value?.data?.message || 'unknown');
+        const firstTrim = errorMessage.substring(jsonFirstIndex + 11);
+        const jsonLastIndex = firstTrim.indexOf('"');
+        const trimmedErrorMessage = firstTrim
+            .substring(0, jsonLastIndex)
```

Comment:
I don't like that I have to call so many `replace` methods. Maybe @valiafetisov you can help me out and generate (or teach me) a rejex expresion to make it a one liner?

Reply:
I'm not sure, what all those replacements do, why do we need them?

Reply:
I wanted to clean up the final error message to remove characters that do not contain any valuable information. 

The first error message would look like this:
`[ethjs-query] while formatting outputs from RPC '{\`, so I removed the special characters at the end to clean it up.